### ISSUE TITLE
Bug: Error on save MeshHex2()

### ISSUE THREAD

**Issue by adtzlr** [kinnala/scikit-fem#639]
Hi,

`mesh.save("out.vtk")` throws an Error for the following lines of code:

```python
skfem.MeshTet2().save('out.vtk')
```

and

```python
skfem.MeshHex2().save('out.vtk')
```

For the case of `MeshHex2` it seems that it is not implemented in the type matching dict in `skfem/io/meshio.py` whereas for the  `MeshTet2` case the element connectivity is of shape `(n, 4)` whereas meshio is expecting `(n, 10)` (n = number of elements).

**Comment by kinnala** [comment]
Btw, looks like VTK does not support quadratic hex mesh in the same sense as we understand it in scikit-fem: it uses different nodes.

**Comment by kinnala** [comment]
Ugh, seems that there is another issue, the constructors of `MeshXXX2` are not really what they are supposed to be. They should use superclass constructor and then call `Mesh.from_mesh`.

**Comment by kinnala** [comment]
Started the fixing work in #640

**Comment by kinnala** [comment]
Work on `MeshHex2` remains. Feel free to fix it using similar steps that in #640 if the feature is needed.
